File: SegundaParte/Semana11/C++/Levantando.cpp
#include <iostream>
#include <string>

class MinhaExcecao {
private:
    int cod;

public:
    MinhaExcecao() : cod(0) {}
    MinhaExcecao(int a) : cod(a) {}

    std::string what() const  {
        return (std::string("Ocorreu a MinhaExcecao com codigo [" + std::to_string(cod) + "]"));
    }
};

int main() {
    try {
        int a = 313;
        if (a > 10) {
            MinhaExcecao minhaExc(a);
            throw minhaExc;
        }
    } catch (const MinhaExcecao& e) {
        std::cout << "Excecao capturada: " << e.what() << std::endl;
    }

    return 0;
}

File: SegundaParte/Semana11/Java/Levantando.java
class MinhaExcecao extends Exception {
  private int cod;
  public MinhaExcecao() {
    cod = 0;
  }
  public MinhaExcecao( int a ) {
    cod = a;
  }
  public String toString() {
    return "Ocorreu a MinhaExcecao com codigo ["+ cod + "]";
  }
}

class Levantando {
  public static void main( String args[]) {
    try {
      int a = 313;
      if( a > 10 ) {
        MinhaExcecao minhaExc = new MinhaExcecao( a );
        throw minhaExc;
      }
    } catch (MinhaExcecao e) {
      System.out.println ("Excecao capturada: " + e);
    }
  }
}

